core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -524,9 +524,16 @@ class GroupMetadataManager(brokerId: Int,
           val memRecords = fetchDataInfo.records match {
             case records: MemoryRecords => records
             case fileRecords: FileRecords =>
-              buffer.clear()
-              fileRecords.readInto(buffer, 0)
-              MemoryRecords.readableRecords(buffer)
+              val sizeInBytes = fileRecords.sizeInBytes
```

Comment:
Can we move the buffer allocation here so that we don't have to reallocate when we need more space?

Reply:
I can definitely see the benefits of moving allocation here, however doing this would require to change from `lazy val buffer` to `var buffer` in [L513](https://github.com/apache/kafka/blob/4a97eafda55c369763fba9a7b3e81aa2f59dd59f/core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala#L513).

This means allocating `buffer` even if it it not needed (in case `fetchDataInfo.records` are `MemoryRecords`, the buffer is not allocated).
That should not be a big deal to have this buffer pre-allocated every time and grow it if needed, just wanted to point out the "downsides" of the approach.
Do you have any opinion @dhruvilshah3 ? 🙂

Reply:
Oh I see, smart! Wouldn’t this be allocating a new buffer at every loop though?